src/torchcodec/_core/AVIOContextHolder.cpp

```diff
@@ -22,13 +23,17 @@ void AVIOContextHolder::createAVIOContext(
       buffer != nullptr,
       "Failed to allocate buffer of size " + std::to_string(bufferSize));
 
-  avioContext_.reset(avio_alloc_context(
+  TORCH_CHECK(
+      (seek != nullptr) && ((write != nullptr) ^ (read != nullptr)),
+      "seek method must be defined, and either write or read must be defined. "
+      "But not both!")
```

Comment:
We may relax the mutual-exclusivity check above eventually, if we implement both `write` and `read` within the same class. For now, mutual-exclusivity is assumed and enforced, because we use the existence of `write` to set the `write_flag` below.